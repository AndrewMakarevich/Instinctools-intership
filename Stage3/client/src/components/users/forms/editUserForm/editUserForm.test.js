import { render, screen } from '@testing-library/react';
import { renderWithRouter } from '../../../../test/helpers/renderWith';
import EditUserForm from './editUserForm';

const userObj = {
  username: 'AndrewTheFirst',
  firstName: 'Andrew',
  lastName: 'Makarevich',
  email: 'andrei1@mail.ru',
};

describe('Correct edit user form', () => {
  test('render', () => {
    renderWithRouter(<EditUserForm userObj={userObj} />);
    expect(screen.getByTestId('edit-user-form')).toBeInTheDocument();
    expect(screen.getByTestId('edit-user-form')).toMatchSnapshot();
  });
});
