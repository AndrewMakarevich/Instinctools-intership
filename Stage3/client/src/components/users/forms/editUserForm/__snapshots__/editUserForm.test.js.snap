// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Correct edit user form render 1`] = `
<form
  data-testid="edit-form"
>
  <div
    class="sc-jSMfEi cgQTUG"
  >
    <label
      class="sc-gsnTZi mhuyQ"
      data-testid="my-label-input"
    >
      <input
        class="sc-dkzDqf fWXHEX"
        placeholder=" "
        value="AndrewTheFirst"
      />
      <span
        class="sc-hKMtZM cnSXuh"
      >
        User's username
      </span>
    </label>
    <label
      class="sc-gsnTZi mhuyQ"
      data-testid="my-label-input"
    >
      <input
        class="sc-dkzDqf fWXHEX"
        placeholder=" "
        value="Andrew"
      />
      <span
        class="sc-hKMtZM cnSXuh"
      >
        User's firstName
      </span>
    </label>
    <label
      class="sc-gsnTZi mhuyQ"
      data-testid="my-label-input"
    >
      <input
        class="sc-dkzDqf fWXHEX"
        placeholder=" "
        value="Makarevich"
      />
      <span
        class="sc-hKMtZM cnSXuh"
      >
        User's lastName
      </span>
    </label>
    <label
      class="sc-gsnTZi mhuyQ"
      data-testid="my-label-input"
    >
      <input
        class="sc-dkzDqf fWXHEX"
        placeholder=" "
        value="andrei1@mail.ru"
      />
      <span
        class="sc-hKMtZM cnSXuh"
      >
        User's email
      </span>
    </label>
  </div>
  <div
    class="sc-eCYdqJ fuUAfF"
  >
    <button
      class="sc-bczRLJ bJUkpm"
      data-testid="clear-changes-btn"
      type="button"
    >
      Clear changes
    </button>
    <button
      class="sc-bczRLJ bJUkpm"
      data-testid="submit-changes-btn"
      type="button"
    >
      Submit changes
    </button>
  </div>
</form>
`;
